Automattic/wp-calypso
# Subscription management: Use subscribe terminology instead of follow

Resolves https://github.com/Automattic/wp-calypso/issues/75973

## Proposed Changes

* use "subscribe" terminology instead of "follow"

## Testing Instructions



1. Check out the PR and build it locally.
2. Get the `subkey` cookie by sending yourself the confirmation email at https://wordpress.com/email-subscriptions and clicking the verification link in the Incognito browser window:

 ![Markup on 2023-03-30 at 11:48:02](https://user-images.githubusercontent.com/25105483/228798142-3b09d1f3-0d7e-40e2-8912-e208919292f6.png)

3. Navigate to http://calypso.localhost:3000/ and set the `subkey` cookie:

![Markup on 2023-03-30 at 11:50:36](https://user-images.githubusercontent.com/25105483/228798729-c31cd569-d69a-45cc-be7f-a6a2d03773f1.png)

4. Make sure the External user you test with, has multiple subscriptions.

If you like, you can use the `https://public-api.wordpress.com/rest/v1.2/read/site/SITE_ID/post_email_subscriptions/new` endpoint to quickly subscribe to multiple sites:

![Markup on 2023-03-30 at 11:44:38](https://user-images.githubusercontent.com/25105483/228797427-f18ae7b7-2c4b-457e-8457-b22ad07e27be.png)

Please note that `SITE_ID` represents the site you are subscribing to and the `Authorization` header needs to match the `subkey` cookie value in the form of `X-WPSUBKEY subkey_value`.

5. Navigate to http://calypso.localhost:3000/subscriptions/ and test for no regressions


_(Credit for the test instructions goes to @ivan-ottinger - https://github.com/Automattic/wp-calypso/pull/75115)_

## Pre-merge Checklist
